classes/Routing/RoutingBootstrapService.php

```diff
@@ -208,4 +215,25 @@ protected function buildEventDispatcher()
 
         return $eventDispatcher;
     }
+
+    /**
+     * Filter the Query Parameter for any Non-Ascii character
+     *
+     * @param Request $request
+     */
+    protected function encodeQueryParams(Request $request): void
+    {
+        $encodedQueryKeys = array_map(function ($requestKeys) {
+            if (preg_match('/%([a-zA-Z].*?)%\d+/', urlencode($requestKeys), $escapeString) == 1) {
```

Comment:
It seems like this regex just matches anything no? If thats the case you might as well do without it and just loop through the params directly

Reply:
Do we need to make sure that we dont affect any key terms also, such as 'relationships', sortBy', orderBy'?

Reply:
Yes, i do take care of it. it will only look for %<serach_tearm>% here. I did try with another normal search & it does work it.

Reply:
I see, missed the '%'. 

I did just try preg_match('/%([a-zA-Z].*?)%\d+/', 'http://avado.local/api/index.php/users?firstnameLIKE%ben%') in the terminal and it doesnt pickanything up?

Reply:
I only seem to get a match with `preg_match('/%([a-zA-Z].*?)%\d+/', 'http://avado.local/api/index.php/users?firstnameLIKE%Ben%2')` which doesnt look to fit the reserved encodings. Im a bit confused :(

Reply:
this won't take full API here, As i am looking only encoded string from query params, So Regex will look like
 preg_match('/%([a-zA-Z].*?)%\d+/', 'firstnameLIKE%Ben%2'), out of which i am looking for "%Ben%2" Value  & wlll replace that string only.

Reply:
shall we call tomorrow? Im not following the `%Ben%2` part.